setup.py

```diff
@@ -372,16 +372,17 @@ def detect_ssl_lib_from_libcurl_dll(self, libcurl_dll_path):
             ssl_lib_detected = 'mbedtls'
         return ssl_lib_detected
 
-    def detect_ssl_lib_on_centos6(self):
+    def detect_ssl_lib_on_centos(self):
         import platform
         from ctypes.util import find_library
         os_name = platform.system()
         if os_name != 'Linux':
             return False
-        dist_name, dist_version, _ = platform.dist()
-        dist_version = dist_version.split('.')[0]
-        if dist_name != 'centos' or dist_version != '6':
-            return False
+        if hasattr(platform, 'dist'):
+            dist_name, dist_version, _ = platform.dist()
+            dist_version = dist_version.split('.')[0]
+            if dist_name != 'centos' or dist_version != '6':
+                return False
```

Comment:
Since this check is now nested in the `hasattr` check, what I believe will happen is OSes other than centos will end up running the subsequent `find_library` code. Instead what should happen is the `find_library` code should only be executed on OSes that have `dist` and whose `dist` is `centos`.

Reply:
yep indeed => PR updated

Reply:
However, no test was failing => any new test to add ?!

Reply:
I think this is not necessary.